Hush/Assets/Scripts/Relics/RelicDome.cs

```diff
@@ -10,21 +12,38 @@ public class RelicDome : MonoBehaviour, IKillable
     {
         [SerializeField] private int keysNeededToUnlock;
 
+        private bool playerIsClose = false;
+
         public UnityEvent attacked;
 
         private UnityEvent _killed;
         public UnityEvent Killed => _killed ??= new UnityEvent();
         
         public float HitPoints => GameManager.Instance.RelicDomeHitPoints;
 
-        private bool CanUnlockDome => GameManager.Instance.KeysInPossession.Count >= keysNeededToUnlock;
+        private bool CanUnlockDome => GameManager.Instance.KeysInPossession.Count >= keysNeededToUnlock && InputManager.Instance.interact && playerIsClose;
+
+        private void Update()
+        {
+            UnlockDome();
```

Comment:
That works but you're doing the check on every frame which is no ideal... You might want to bind the performed action of the Interact button so that you only evaluate if the player is close to the relic when you press F...

Same can be done for doors!

Reply:
^ agree